### PR TITLE
Aromaticity and Gasteiger charges guessers

### PR DESCRIPTION
**Pull request by cbouy**
Part of #2468

Changes made in this Pull Request:
 - Add a guesser for the *aromaticities* `TopologyAttr`
 - Add a guesser for partial charges (Gasteiger)

PR Checklist
------------
 - [x] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

**Comment by cbouy** [comment]
@richardjgowers @IAlibay @fiona-naughton Not sure who I'm supposed to ping now that GSoC ended 😅

File: package/MDAnalysis/topology/guessers.py [hunk 1]
```diff
@@ -475,3 +475,48 @@ def guess_atom_charge(atomname):
...
+
+    .. versionadded:: 2.0.0
+    """
+    mol = atomgroup.convert_to("RDKIT")
+    atoms = sorted(mol.GetAtoms(),
+                   key=lambda a: a.GetIntProp("_MDAnalysis_index"))
+    return np.array([atom.GetIsAromatic() for atom in atoms])
+
+
+def guess_gasteiger_charges(atomgroup):
```
Comment (IAlibay) [inline package/MDAnalysis/topology/guessers.py:501]: I think for now it's fine, although long term we might want to consider how we want this to look API-wise. In a hypothetical world where we have converters for things like openbabel, we might end up with multiple possible charge assignment methods. It'd be nice to have them all accessible from the same entry point.

**Review by IAlibay — CHANGES_REQUESTED** [review]
Overall lgtm, but needs tests and usage docs (there's some example use docstring at the top that could be good to add to, although we might want to migrate to user guide, @lilyminium thoughts?).

Couple of comments, there's a long term thought about how we want our guessers to look like in a multi-converter world. Overall we might need to completely overhaul the guessers, so I'm not convinced it needs to be addressed here.

**Comment by IAlibay** [comment]
Forgot to add; can't speak for everyone, but I don't mind if you keep pinging me for the RDKit things. I get the email notifications whether or not you do so anyways :)

File: package/MDAnalysis/topology/guessers.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (cbouy) [reply package/MDAnalysis/topology/guessers.py:501]: As far as I understand it yes, it only looks at the connectivity

**Comment by IAlibay** [comment]
Slowly going back into the RDKIT stuff.

@cbouy if you're still interested in progressing here, I think what's stalling the PR is the lack of tests.

**Review by IAlibay — APPROVED** [review]
This looks fine to me. @lilyminium can you have a glance at this?

**Review by lilyminium — APPROVED** [review]
LGTM. I am curious what the performance would be like for a molecule from a non-rdkit (e.g. PDB) format, but problems deriving aromaticity/charges should be solved in the RDKitConverter I would say.

**Comment by IAlibay** [comment]
Thanks @cbouy ! -- squash merging

**Comment by IAlibay** [comment]
😱 We forgot the changelog!
@cbouy can you add a changelog entry for this in #2942?

**Pull request merged**